packages/salesforcedx-utils-vscode/src/cli/commandOutput.ts

```diff
@@ -9,13 +9,22 @@ import { CommandExecution } from './commandExecutor';
 
 export class CommandOutput {
   public async getCmdResult(execution: CommandExecution): Promise<string> {
+    let buffer = '';
     return new Promise<
       string
     >((resolve: (result: any) => void, reject: (reason: string) => void) => {
       execution.processExitSubject.subscribe(data => {
         if (data != undefined && data.toString() === '0') {
```

Comment:
@GaryFNC @chowwinston - I _think_ I know why we are seeing the hangs or incomplete data. If you do it this way, I suspect that you are only draining data from `execution.stdoutSubject.subscribe` at the end after you have received an `execution.processExitSubject`. However, by that time, the stdout buffer has already filed up and the process is stuck from not draining it.

I know I had suggested serializing the result so that we only get them on processExit but this probably prevents the  data from being drained.

Reply:
As a quick sanity check, https://nodejs.org/api/child_process.html#child_process_event_exit says:

>The 'exit' event is emitted after the child process ends. If the process exited, code is the final exit code of the process, otherwise null. If the process terminated due to receipt of a signal, signal is the string name of the signal, otherwise null. One of the two will always be non-null. **Note that when the 'exit' event is triggered, child process stdio streams might still be open.**

And, https://nodejs.org/api/child_process.html#child_process_event_close says:

>**The 'close' event is emitted when the stdio streams of a child process have been closed.** This is distinct from the 'exit' event, since multiple processes might share the same stdio streams.

So, instead of listening on the exit event, we might want to listen to the _close_ event when data from the streams are important.

So, things to try:

1. Don't serialize the promise to only read from stdout and stderr if the exit signal has been received. Instead, continue reading and buffer the chunks.
1. Instead of listening to _exit_, listen to _close_ to know when all data from stdio have been received, then you can proceed to parse the JSON, once and for all (you no longer need the weird trick of parsing each time).

Reply:
Actually a bit of spelunking shows that the serialization technique was done earlier as part of https://github.com/forcedotcom/salesforcedx-vscode/commit/8c3cebda40148e30f451aa471309a9a5cca6b971#diff-222256c5627f520a77b31ab383f6ffa1. When you have small responses, this doesn't present a problem, but when you have larger responses, this becomes an issue.